languagetool-core/src/main/java/org/languagetool/rules/spelling/morfologik/MorfologikSpellerRule.java

```diff
@@ -220,7 +221,13 @@ protected Translator getTranslator(GlobalConfig globalConfig) {
     return null;
   }
 
-  private boolean initSpellers() throws IOException {
+  private void initSpellersWithLock() throws IOException {
```

Comment:
Why do you need a separate withLock method? Can you add the lock inside initSpellers?